### ISSUE TITLE
Portable Mode - Early Feedback

### ISSUE THREAD

**Issue by seanmcbreen** [Microsoft/vscode#52656]
Issue Type: <b>Bug</b>

First off I **love the progress** here - this does work and it works well.  That said I wanted to create an item with some feedback I had from playing with this on a few machines.

# Slow USB Device

I used a very slow USB device for my testing - it did work but many of they things I had grown used to in VS Code felt slow.  For instance... 
- Installing an extension, 
- Expanding an Emmet Abbreviation, 
- Opening a Markdown Preview 

Generally all code loading :)

## Suggestions
1. Can we provide some guidelines [through documentation] on a USB specification we fell good about from a perfr perspective.
2. Can we detect the spec of the USB device - and if it's below our ideal spec can we send a notification to the end user that some experiences are degraded.
3. When we are waiting on code loading (I'm assuming)

# USB Pulled out while running
I saw a few errors here when the files could not be saved - which essentially prevented VS Code from closing.

## Suggestions
1. Could we explain a little more the issue for saving as an example.

# Guidelines for creation
Really another version of the above - can we suggest the best path in the docs to create an image.  I see two uses cases:

1. A common image - I would create first on the local HHD then export as a zip or USB image to others.  In that process I would make setting changes, extension installs and other items before moving to USB.
2. An individual image - this is interesting e.g. people chat about using drop box or one drive.  Then we I think should suggest an optimal path to that.

# System Information 

It would be good to be clear when a device/install is operating in portable mode.   For instance I'm reporting this issue form my portable install and you can see that by looking at the folder name but we might want additional information e.g. 
- Install Type: Portable
- Drive Details: SSD Speed/Specs

Sorry a little messy as an 'issue' but I do really like the progress and see a ton of potential.  I still have my USB in pocket right now :)

VS Code version: Code - Insiders 1.25.0-insider (b4a18da6e78f95295ea6538f5ae8dd0c9f0869a9, 2018-06-19T07:49:07.900Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz (4 x 2808)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.93GB (10.00GB free)|
|Process Argv|D:\VS Code Portable\code-inisders\Code - Insiders.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
eclipse-new-java-project|mel|1.0.0
vsliveshare|ms-|0.3.295
java|red|0.27.0
vscode-java-debug|vsc|0.9.0
vscode-java-pack|vsc|0.3.0
vscode-java-test|vsc|0.6.1
vscode-maven|vsc|0.9.0
Issue Type: <b>Bug</b>

First off I love the proigress here - this does work and it works well.  That said I wanted to create an item with some feedback I had from playing with this on a few machines.

# Slow USB Device

I used a very slow USB device for my testing - it did work but many of they things I had grown used to in VS Code felt slow.  For instance... 
- Installing an extension, 
- Expanding an Emmet Abreviation, 
- Opening a Markdown Preview 

Gernerally all code loading :)

## Suggestions
1. Can we provide some guidelines [through documentation] on a USB specification we fell good about from a perf perspective.
2. Can we detect the spec of the USB device - and if it's below our ideal spec can we send a notification to the end user that some experiences arte degraded.
3. When we are waiting on code loading (I'm assuming)

# USB Pulled out while running
I saw a few errors here when the files could not be saved - which essentially prevented VS Code from closeing.

## Suggestions
1. Could we explain a little more the issue for saving as an example.
2. Is it possible to know the USB was removed (our mounted dfrive etc).

# Guidelines for creation
Really anotyher version of hte above - can we suggest the best path in the docs to create an image.  I see two uses cases:

1. A common image - I would create fiurst on the locall HHD then export as a zip or USB image to others.  In that process I would make setting schanges, extension installs and other items before moving to USB.
2. An indivdual image - this is interesting e.g. people chat about using drop box or one drive.  Then we I think should sugest an optimal path to that.

# System Information 

It would be good to be clear when a device/install is operating in portable mode.   For instance I'm reporting this issue form my portable install and you can see thet by looking at the folder name but we might want tadditional information e.g. 
- Install Type: Portable
- Drive Details: SSD Speed/Specs

Sorry a little messy as an 'issue' but I do really like the progress and see a ton of potential.  I still have my USB in pocket right now :)

VS Code version: Code - Insiders 1.25.0-insider (b4a18da6e78f95295ea6538f5ae8dd0c9f0869a9, 2018-06-19T07:49:07.900Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz (4 x 2808)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.93GB (10.00GB free)|
|Process Argv|D:\VS Code Portable\code-inisders\Code - Insiders.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
eclipse-new-java-project|mel|1.0.0
vsliveshare|ms-|0.3.295
java|red|0.27.0
vscode-java-debug|vsc|0.9.0
vscode-java-pack|vsc|0.3.0
vscode-java-test|vsc|0.6.1
vscode-maven|vsc|0.9.0

</details>

</details>

**Comment by joaomoreno** [comment]
Portable mode isn't anything other than the regular old ZIP with different defaults for the `--user-data-dir` and `--extensions-dir` flags. It enables data locality.

Officially standing behind the claim that it will run fine off a USB drive will be a support nightmare IMO.

> An individual image - this is interesting e.g. people chat about using drop box or one drive. Then we I think should suggest an optimal path to that.

I would especially refrain from suggesting placing VS Code and data in Dropbox/OneDrive. This will definitely not work nicely.

**Comment by seanmcbreen** [comment]
I agree with those points - so I think it really simply comes down to providing solid guidance and setting expectations crisply.  We should state the scenarios this opens up but also which ones we don't recommend.

Question - are we thinking that this would replace the existing ZIP install - if so should we do the peer folder creation in that zip?  Perhaps that would be a migration issue for existing users.  It's also a little different for say the Mac/Linux user.  

My real question is should this replace an existing install option and be simply config for say the apple user and be set-up as default for the Windows Zip user.

**Comment by joaomoreno** [comment]
> Question - are we thinking that this would replace the existing ZIP install - if so should we do the peer folder creation in that zip? Perhaps that would be a migration issue for existing users. It's also a little different for say the Mac/Linux user.

It's an interesting thought. Though I think there are still users who don't mind having their data in the default system location while still using the ZIP as a distribution. I think switching the default behaviour of the ZIP would result in some confusion... It's just a gut feelign tho...

**Comment by seanmcbreen** [comment]
OK good point - our lists of builds is getting big (to maintain and also to direct users to).  My gut for this one based on your comment is to only link it from the 'portable-mode' page you authored.  Does that work - if so close :)

**Comment by Ciantic** [comment]
The documentation should be more clear that VSCode in portable mode can't update itself automatically. ( https://stackoverflow.com/questions/50163688/how-to-update-visual-studio-code-portable )

Basically installing a real version and making directory junctions for those two directories is a better option than portable mode.
